src/main/java/io/lettuce/core/LMoveArgs.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2011-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.lettuce.core;
+
+import io.lettuce.core.internal.LettuceAssert;
+import io.lettuce.core.protocol.CommandArgs;
+
+import static io.lettuce.core.protocol.CommandKeyword.*;
+
+/**
+ * Argument list builder for the Redis <a href="http://redis.io/commands/blmove">BLMOVE</a> and
+ * <a href="http://redis.io/commands/lmove">LMOVE</a> commands. Static import the methods from {@link Builder} and
+ * chain the method calls: {@code source(LEFT).destination(RIGHT)}.
+ *
+ * @author Mikhael Sokolov
+ */
+public class LMoveArgs implements CompositeArgument {
+
+    public enum Direction {
```

Comment:
Let's keep the enum `private`. Redis allows either `LEFT RIGHT` or `RIGHT LEFT`. Let's align the enum constants in a similar way (`LEFT_RIGHT`, `RIGHT_LEFT`). Doing so reduces the number of invariants from 4 to 2. Once the command changes, then we have the freedom to update our enum without breaking code that uses this args class.

Reply:
@mp911de That's was actually my first thought when I was wondering of how to implement it in a best way (smth like ltr and rtl static methods or stuff like that).
But the truth is that i could be also either RIGHT RIGHT or LEFT LEFT (you can move element from one head not just to another list tail, you are able to insert it in the head of it).
![image](https://user-images.githubusercontent.com/21218309/98179566-178d3500-1f10-11eb-8ad5-19122d90ab59.png)
So i do still think that it is an appropriate API args draft.

Reply:
what about like this:
```
public static LMoveArgs move(Direction source, Direction dest) {
//...
}
```
The `source ` and `dest ` are required.